docs/grpc/java.md

```diff
@@ -125,37 +125,33 @@ import java.nio.file.Paths;
 import java.util.concurrent.Executor;
 
 public class Main {
-  static class MacaroonCallCredential implements CallCredentials {
-    private final String macaroon;
+    static class MacaroonCallCredential extends CallCredentials {
+        private final String macaroon;
 
-    MacaroonCallCredential(String macaroon) {
-      this.macaroon = macaroon;
-    }
+        MacaroonCallCredential(String macaroon) {
+            this.macaroon = macaroon;
+        }
+
+        @Override
+        public void applyRequestMetadata(RequestInfo requestInfo, Executor executor, MetadataApplier metadataApplier) {
+            executor.execute(() -> {
+                try {
+                    Metadata headers = new Metadata();
+                    Metadata.Key<String> macaroonKey = Metadata.Key.of("macaroon", Metadata.ASCII_STRING_MARSHALLER);
+                    headers.put(macaroonKey, macaroon);
+                    metadataApplier.apply(headers);
+                } catch (Throwable e) {
+                    metadataApplier.fail(Status.UNAUTHENTICATED.withCause(e));
+                }
+            });
+        }
 
-    public void thisUsesUnstableApi() {}
-
-    public void applyRequestMetadata(
-        MethodDescriptor < ? , ? > methodDescriptor,
-        Attributes attributes,
-        Executor executor,
-        final MetadataApplier metadataApplier
-    ) {
-      String authority = attributes.get(ATTR_AUTHORITY);
-      System.out.println(authority);
-      executor.execute(new Runnable() {
-        public void run() {
-          try {
-            Metadata headers = new Metadata();
-            Metadata.Key < String > macaroonKey = Metadata.Key.of("macaroon", Metadata.ASCII_STRING_MARSHALLER);
-            headers.put(macaroonKey, macaroon);
-            metadataApplier.apply(headers);
-          } catch (Throwable e) {
-            metadataApplier.fail(Status.UNAUTHENTICATED.withCause(e));
-          }
+        @Override
+        public void thisUsesUnstableApi() {
         }
-      });
+
```

Comment:
nit: extra newlines

Reply:
Thanks for your advice. I have made another modification to the document